tests/integration/tasks/sync.ts

```diff
@@ -1,10 +1,53 @@
+import * as chalk from 'chalk';
 import { handler } from '../../../src/commands/tasks/sync';
 import { runtimeChoices } from '../../../src/constants';
+import { functionCode } from '../../__helpers__/functions';
+import { tasksRepositoryMock } from '../../__helpers__/tasksRepositoryMock';
+import { createTempDirectoryManager } from '../../__helpers__/tempDirectoryManager';
 
 describe('exh tasks sync', () => {
+  let tempDirectoryManager;
+  let repositoryMock;
+
+  beforeEach(async () => {
+    tempDirectoryManager = await createTempDirectoryManager();
+  });
+
+  afterEach(async () => {
+    jest.clearAllMocks();
+    await tempDirectoryManager.removeDirectory();
+  });
+
   const root = 'tests/__helpers__/task-configs/invalid-runtimes/';
   const runtimes = runtimeChoices.map(runtime => runtime).join(', ');
 
+  it('Creates a Function', async () => {
+    repositoryMock = tasksRepositoryMock();
+    const taskConfigPath = await tempDirectoryManager.createTempJsonFile(repositoryMock.functionConfig);
+    await tempDirectoryManager.createTempJsFile('index', functionCode);
+
+    const logSpy = jest.spyOn(global.console, 'log');
+
+    await handler({ sdk: null, path: taskConfigPath });
+    expect(repositoryMock.findFunctionsSpy).toHaveBeenCalledTimes(1);
+    expect(repositoryMock.createFunctionSpy).toHaveBeenCalledTimes(1);
+    expect(logSpy).toHaveBeenCalledWith(chalk.green('Successfully created task', repositoryMock.functionConfig.name));
+  });
+
+  it('Updates a Function', async () => {
+    repositoryMock = tasksRepositoryMock(true);
```

Comment:
Don't really like booleans as a  parameter like this because they are non-descriptive. I had to go look tot the code of the `tasksRepositoryMock`.  I would prefer to just use `mockImplementationOnce` on `tasksRepositoryMock.find` like we did in the `Updates an existing Dispatcher` test.

Reply:
I've changed it, but I think it makes the test way more complex that it was before